library/L2_HAL/actuators/servo/servo.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+
+#include "L1_Peripheral/pwm.hpp"
+#include "utility/map.hpp"
+
+namespace sjsu
+{
+class Servo
+{
+ public:
+    static constexpr uint32_t kDefaultFrequency = 50;
+    static constexpr float kMinAngle = 0;
+    static constexpr float kMaxAngle = 360;
+    static constexpr float kMinPulse = 500;
+    static constexpr float kMaxPulse = 2500;
+    explicit constexpr Servo(const sjsu::Pwm & pwm)
+      : servo_pwm_(pwm),
+        max_pulse_length_us_(0),
+        pulse_lower_bound_(kMinPulse),
+        pulse_upper_bound_(kMaxPulse),
+        min_angle_(kMinAngle),
+        max_angle_(kMaxAngle)
+    {
+    }
+
+    virtual void Initialize(uint32_t frequency = kDefaultFrequency)
+    {
+      servo_pwm_.Initialize(frequency);
+      max_pulse_length_us_ = (1 / static_cast<float>(frequency)) * 1'000'000;
+    }
+
+    virtual void SetFrequency(uint32_t frequency = kDefaultFrequency)
+    {
+      servo_pwm_.SetFrequency(frequency);
+      max_pulse_length_us_ = (1 / static_cast<float>(frequency)) * 1'000'000;
+    }
+
+    virtual void SetPulseBounds(float lower, float upper)

```

Comment:
Although the name is pretty intiuative, I believe that a comment here will better explain that the pulse bound is the lowest pulse with that this servo object will allow. Meaning passing a lower or higher pulse width will clamp to to these bounds. Note in this that the lower bounds should be what constitutes a 0 degree and the higher should be the highest degree.